packages/@uppy/transloadit/src/index.ts

```diff
@@ -697,12 +699,13 @@ export default class Transloadit<
     // Set up the Assembly instances and AssemblyWatchers for existing Assemblies.
     const restoreAssemblies = () => {
       this.#createAssemblyWatcher(previousAssembly.assembly_id)
-      this.#connectAssembly(this.assembly!)
+      if (this.assembly == null) throw new Error('Assembly was nullish')
+      this.#connectAssembly(this.assembly)
```

Comment:
How is that better?

Reply:
Let's move that to a separate PR to discuss how to solve it properly. IMO throwing a `Assembly was nullish` error instead of a generic `TypeError` is not an improvement, we might as well keep the current non-nullish assertion.
```suggestion
      this.#connectAssembly(this.assembly!)
```